File: 프로그래머스/0/181866. 문자열 잘라서 정렬하기/문자열 잘라서 정렬하기.java
import java.util.Arrays;
import java.util.ArrayList;

class Solution {
    public String[] solution(String myString) {
        ArrayList<String> list = new ArrayList<>();
        String str = "";
        for (char c : myString.toCharArray()) {
            if (c == 'x') {
                if (!str.isEmpty()) {
                    list.add(str);
                    str = "";
                }
            } else {
                str += c;
            }
        }
        if(!str.isEmpty()) list.add(str);
        String answer[] = list.toArray(new String[list.size()]);
        Arrays.sort(answer);
        return answer;
    }
}

File: 프로그래머스/0/181866. 문자열 잘라서 정렬하기/문자열 잘라서 정렬하기.js
function solution(myString) {
    let answer = [];
    let str = "";
    for (let c of myString) {
        if (c == 'x') {
            if (str != "") {
                answer.push(str);
                str = "";
            }
        } else {
            str += c;
        }
    }
    if(str != "") answer.push(str);
    return answer.sort();
}